pulser/tests/test_register_layout.py

```diff
@@ -111,3 +118,64 @@ def test_eq():
     layout2 = RegisterLayout([[1, 0], [0, 0]])
     assert layout1 == layout2
     assert hash(layout1) == hash(layout2)
+
+
+def test_traps_from_coordinates():
+    assert layout._coords_to_traps == {
+        (0, 0): 0,
+        (0, 1): 1,
+        (1, 0): 2,
+        (1, 1): 3,
+    }
+    assert layout.get_traps_from_coordinates(
+        (0.9999995, 0.0000004), (0, 1), (1, 1)
+    ) == [2, 1, 3]
+    with pytest.raises(ValueError, match="not a part of the RegisterLayout"):
+        layout.get_traps_from_coordinates((0.9999994, 1))
+
+
+def test_square_lattice_layout():
+    square = SquareLatticeLayout(9, 7, 5)
+    assert str(square) == "SquareLatticeLayout(9x7, 5µm)"
+    assert square.square_register(3) == Register.square(
+        3, spacing=5, prefix="q"
+    )
+    # An even number of atoms on the side won't align the center with an atom
```

Comment:
I'm not sure to understand the comment: is its intent to explain why registers are different on lines 144 and 145? If so, the used sizes `3` and `4` are different, which is already an explanation.

Reply:
You're right, this is a typo, it should be `4` on both